lib/logstasher/rails_ext/action_controller/metal/instrumentation.rb

```diff
@@ -23,27 +23,28 @@ def process_action(*args)
           logstasher_add_custom_fields_to_request_context(LogStasher.request_context)
         end
 
-        if self.respond_to?(:logtasher_add_custom_fields_to_payload)
-          before_keys = raw_payload.keys.clone
-          logtasher_add_custom_fields_to_payload(raw_payload)
-          after_keys = raw_payload.keys
-          # Store all extra keys added to payload hash in payload itself. This is a thread safe way
-          LogStasher.custom_fields += after_keys - before_keys
+        begin
+          result = super
+          payload[:status] = response.status
+          result
+        ensure
+          if self.respond_to?(:logtasher_add_custom_fields_to_payload)
```

Comment:
i think it should be `:logstasher_add_custom_fields_to_payload` instead of `:logtasher_add_custom_fields_to_payload`